stl/inc/span

```diff
@@ -414,23 +415,38 @@ public:
     constexpr span(const array<value_type, _Size>& _Arr) noexcept : _Mybase(_Size), _Mydata(_Arr.data()) {}
 
     template <class _Rng>

```

Comment:
We can change this to `template <_Is_span_compatible_Range<element_type> _Rng>` and drop the requires-clause on line 418.

Reply:
@BillyONeal wont like it but I love concepts

Reply:
@miscco
* I only dislike concepts insofar as they encourage designing interfaces that require SFINAE tricks, and I expect users to shoot themselves in the foot with that. If you're considering something that relies on SFINAE tricks anyway they're an improvement over the equivalent enable_if spew.
* I dislike defining interfaces that rely heavily on SFINAE, because the diagnostics for SFINAE are terrible / nonexistent. For example, the other day I messed up and forgot one of the typedefs necessary for an iterator. Instead of "whoops, I needed this type and you didn't give me one", the iterator_traits type became empty, which completely disabled the overload set expecting iterators I was looking for. And so you get diagnostics like "couldn't convert `test_iterator` to `size_t` and no indication that there was an iterator pair version you were trying to call.
* I dislike concepts' proponents arguing that they produce better diagnostics, given substantial implementation experience suggesting that they do not. Certain high profile C++ folks say things like "I want a diagnostic saying XYZ isn't sortable", suggest concepts as a solution, and fail to acknowledge that concepts have delivered nothing of the sort.